code/modules/cargo/packs/organic.dm

```diff
@@ -446,3 +440,11 @@
 					/obj/item/valentine)
 	crate_name = "valentine crate"
 	crate_type = /obj/structure/closet/crate/secure
+
+//////////////Special Code for Special Crates//////////////
+
+/datum/supply_pack/organic/randomizedcandy/fill(obj/structure/closet/crate/C)
+	var/list/L = contains.Copy()
+	for(var/i in 1 to num_contained)
+		var/item = pick_n_take(L)
+		new item(C)
```

Comment:
Aaaand remove this once you repathed the candies supply pack.

On a sidenote: Don't do ever do this again without good reason and keep these niche proc overrides next to their own types, you'll only make reviewing harder this way.

Reply:
To be fair, this is the way it was before I started messing with it, I only started moving stuff around to try and fix ordering issues.